test/gurka/test_instructions_internal_intersection.cc

```diff
@@ -0,0 +1,117 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+
+class InstructionsInternalIntersection : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize_metres = 10;
+
+    const std::string ascii_map = R"(
+              E D
+              | |
+              | |
+              | |
+              | |
+         L----F-C----K
+              | |
+         I----G-B----J
+              | |
+              | |
+              | |
+              | |
+              H A
+    )";
+
+    const gurka::ways ways =
+        {{"AB", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"BC", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"CD", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"EF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"FG", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"GH", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Broken Land Parkway"}}},
+         {"IG", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}},
+         {"GB", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"BJ", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"KC", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"CF", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Snowden River Parkway"}}},
+         {"FL", {{"highway", "primary"}, {"oneway", "yes"}, {"name", "Patuxent Woods Drive"}}}};
+
+    const auto layout =
+        gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+
+    map =
+        gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_instructions_internal_intersection",
+                          {{"mjolnir.admin",
+                            {VALHALLA_SOURCE_DIR "test/data/netherlands_admin.sqlite"}}});
+  }
+};
+
+gurka::map InstructionsInternalIntersection::map = {};
+
+///////////////////////////////////////////////////////////////////////////////
+// Left turn
+TEST_F(InstructionsInternalIntersection, BrokenLandParkway_TurnLeft_PatuxentWoodsDrive) {
```

Comment:
Can these tests also assert the begin/end location of the maneuvers?

Reply:
you mean index instead of the distance?

Reply:
Right, yeah.

Reply:
@danpaz I added index tests